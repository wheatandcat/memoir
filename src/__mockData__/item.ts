import { ItemQuery } from 'queries/api/index';
type Item = NonNullable<ItemQuery['item']>;

export const item = (option?: Partial<Item>): Item => ({
  id: '1',
  title: '買い物',
  categoryID: 9,
  date: '2021-01-01T00:00:00+09:00',
  like: false,
  dislike: false,
  createdAt: '2021-01-01T00:00:00+09:00',
  updatedAt: '2021-01-01T00:00:00+09:00',
  ...option,
});

export const items = (): Item[] => [
  item(),
  item({
    id: '2',
    title: '本を読む',
    categoryID: 2,
    date: '',
    dislike: false,
    like: false,
  }),
];
