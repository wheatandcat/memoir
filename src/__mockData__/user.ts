import { User } from 'queries/api/index';

export const user = (option?: Partial<User>): User => ({
  id: '1',
  displayName: 'test',
  image: '',
  updatedAt: '2021-01-01T00:00:00+09:00',
  createdAt: '2021-01-01T00:00:00+09:00',
  ...option,
});

export const users = (): User[] => [
  {
    id: 'test1',
    displayName: 'suzuki1',
    image: 'https://placehold.jp/150x150.png',
    updatedAt: '2021-01-01T00:00:00+09:00',
    createdAt: '2021-01-01T00:00:00+09:00',
  },
  {
    id: 'test2',
    displayName: 'suzuki2',
    image: 'https://placehold.jp/150x150.png',
    updatedAt: '2021-01-01T00:00:00+09:00',
    createdAt: '2021-01-01T00:00:00+09:00',
  },
  {
    id: 'test3',
    displayName: 'suzuki4',
    image: 'https://placehold.jp/150x150.png',
    updatedAt: '2021-01-01T00:00:00+09:00',
    createdAt: '2021-01-01T00:00:00+09:00',
  },
  {
    id: 'test4',
    displayName: 'suzuki4',
    image: 'https://placehold.jp/150x150.png',
    updatedAt: '2021-01-01T00:00:00+09:00',
    createdAt: '2021-01-01T00:00:00+09:00',
  },
  {
    id: 'test5',
    displayName: 'suzuki5',
    image: 'https://placehold.jp/150x150.png',
    updatedAt: '2021-01-01T00:00:00+09:00',
    createdAt: '2021-01-01T00:00:00+09:00',
  },
];
