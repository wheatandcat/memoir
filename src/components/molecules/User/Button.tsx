import React, { memo } from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import View from 'components/atoms/View';
import { User } from 'queries/api/index';
import UserImage from 'components/molecules/User/Image';

export type Props = {
  user: Pick<User, 'id' | 'image'>;
  selected: boolean;
  onAdd: (id: User['id']) => void;
  onRemove: (id: User['id']) => void;
  size?: number;
};

const Users: React.FC<Props> = (props) => {
  if (props.selected) {
    return (
      <View m={2}>
        <TouchableOpacity onPress={() => props.onRemove(props.user.id)}>
          <UserImage size={props.size} image={props.user.image} />
        </TouchableOpacity>
      </View>
    );
  }

  return (
    <View m={2} style={styles.clear}>
      <TouchableOpacity onPress={() => props.onAdd(props.user.id)}>
        <UserImage size={props.size} image={props.user.image} />
      </TouchableOpacity>
    </View>
  );
};

Users.defaultProps = {
  size: 70,
};

export default memo(Users);

const styles = StyleSheet.create({
  clear: {
    opacity: 0.4,
  },
});
