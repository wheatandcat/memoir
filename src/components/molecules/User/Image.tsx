import React, { memo } from 'react';
import { StyleSheet, ImageProps } from 'react-native';
import Image from 'components/atoms/Image';
import theme from 'config/theme';

export type Props = {
  image: string | null;
  size?: number;
  onLoadEnd?: ImageProps['onLoadEnd'];
};

const UserImage: React.FC<Props> = (props) => {
  return (
    <>
      {props.image ? (
        <Image
          source={{ uri: props.image }}
          style={styles.image}
          width={props.size}
          height={props.size}
          contentFit="cover"
          onLoadEnd={props.onLoadEnd}
        />
      ) : (
        <Image
          source={require('../../../img/icon/icon_account_default.png')}
          width={props.size}
          height={props.size}
          style={styles.noImage}
          onLoadEnd={props.onLoadEnd}
          testID="default-user-image"
        />
      )}
    </>
  );
};

const styles = StyleSheet.create({
  image: {
    borderRadius: 60,
    borderWidth: StyleSheet.hairlineWidth,
    borderColor: theme().color.base.main,
  },
  noImage: {
    backgroundColor: theme().color.background.light,
    borderRadius: 60,
  },
});

UserImage.defaultProps = {
  size: 100,
};

export default memo(UserImage);
