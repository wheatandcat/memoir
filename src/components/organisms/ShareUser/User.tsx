import React, { memo, useCallback } from 'react';
import { StyleSheet, TouchableOpacity, Alert } from 'react-native';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import UserImage from 'components/molecules/User/Image';
import { User as UserType } from 'store/atoms';
import theme from 'config/theme';
import { ConnectedType } from 'components/pages/MyPage/Connected';

export type Props = {
  loading: boolean;
  user: UserType;
  onDeleteRelationship: ConnectedType['onDeleteRelationship'];
};

const User: React.FC<Props> = (props) => {
  const onDelete = useCallback(() => {
    Alert.alert('共有メンバーを解除しますか？', '', [
      {
        text: 'キャンセル',
        style: 'cancel',
      },
      {
        text: '解除する',
        onPress: () => {
          props.onDeleteRelationship(props.user.id || '');
        },
      },
    ]);
  }, [props]);

  return (
    <View style={styles.root}>
      <View>
        <UserImage image={props.user?.image || ''} size={100} />
      </View>
      <View>
        <View my={2}>
          <Text textAlign="center">{props.user?.displayName || ''}</Text>
        </View>
        <TouchableOpacity
          onPress={() => {
            if (props.loading) {
              return;
            }

            onDelete();
          }}
        >
          <Text size="xs" textAlign="center">
            解除
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: theme().space(3),
    width: 110,
  },
});

export default memo(User);
