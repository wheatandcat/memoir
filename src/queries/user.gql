query User {
  user {
    id
    displayName
  }
}
