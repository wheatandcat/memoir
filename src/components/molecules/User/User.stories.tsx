import React from 'react';
import Image, { Props } from './Image';

const props = (): Props => ({ image: null });

export default {
  title: 'molecules/User',
};

export const _Image = () => <Image {...props()} />;
