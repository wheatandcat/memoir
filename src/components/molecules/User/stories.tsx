import React from 'react';
import { storiesOf } from '@storybook/react-native';
import Image, { Props } from './Image';

const props = (): Props => ({ image: null });

storiesOf('molecules/User', module).add('Image', () => <Image {...props()} />);
