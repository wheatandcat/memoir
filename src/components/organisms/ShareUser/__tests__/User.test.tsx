import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { user } from '__mockData__/user';
import User, { Props } from '../User';

const propsData = (): Props => ({
  user: user(),
  loading: false,
  onDeleteRelationship: jest.fn(),
});

describe('components/organisms/ShareUser/User.tsx', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<User {...propsData()} />);
  });

  it('正常にrenderすること', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
