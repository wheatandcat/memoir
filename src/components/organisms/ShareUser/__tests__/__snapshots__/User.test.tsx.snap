// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/organisms/ShareUser/User.tsx 正常にrenderすること 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginBottom": 16,
      "width": 110,
    }
  }
>
  <View>
    <Memo(UserImage)
      image=""
      size={100}
    />
  </View>
  <View>
    <View
      my={2}
    >
      <Text
        fontFamily="RobotoCondensed-Bold"
        textAlign="center"
        variants="body"
      >
        test
      </Text>
    </View>
    <TouchableOpacity
      onPress={[Function]}
    >
      <Text
        fontFamily="RobotoCondensed-Bold"
        size="xs"
        textAlign="center"
        variants="body"
      >
        解除
      </Text>
    </TouchableOpacity>
  </View>
</View>
`;
